pkg/core/blockchain.go

```diff
@@ -1622,7 +1625,7 @@ func (bc *Blockchain) storeBlock(block *block.Block, txpool *mempool.Pool) error
 				stop = start + 1
 			}
 			for index := start; index < stop; index++ {
-				err := kvcache.DeleteBlock(bc.GetHeaderHash(index))
+				err := kvcache.DeleteBlock(bc.GetHeaderHash(index), bc.config.Ledger.RemoveUntraceableHeaders)
```

Comment:
P2P can deal with it, you just don't get anything for your request, it can happen on a P2P network. Additional capabilities would make it much better, but they're not mandatory for the scheme to work.